pkg/cloud/tags/tags_test.go

```diff
@@ -94,3 +129,73 @@ func TestTags_ComputeDiff(t *testing.T) {
 		})
 	}
 }
+
+func TestTags_Ensure(t *testing.T) {
+	tests := []struct {
+		name          string
+		builder       Builder
+		builderOption BuilderOption
+		expect        func(m *mock_ec2iface.MockEC2APIMockRecorder)
+	}{
+		{
+			name:    "Should return error when applyFunc params fails",
+			builder: Builder{params: &bp},
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.CreateTags(gomock.Eq(&ec2.CreateTagsInput{
+					Resources: aws.StringSlice([]string{""}),
+					Tags:      tags,
+				})).Return(nil, errors.New("failed to create tag"))
+			},
+		},
+		{
+			name:          "Should return error when applyFunc is nil",
+			builder:       Builder{params: &bp, applyFunc: nil},
+			builderOption: func(builder *Builder) {},
+		},
+		{
+			name:          "Should return error when build params is nil",
+			builder:       Builder{params: nil},
+			builderOption: func(builder *Builder) {},
+		},
+		{
+			name:          "Should ensure tags successfully",
+			builder:       Builder{params: &bp},
+			builderOption: func(builder *Builder) {},
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.CreateTags(gomock.Eq(&ec2.CreateTagsInput{
+					Resources: aws.StringSlice([]string{""}),
+					Tags:      tags,
+				})).Return(nil, nil)
+			},
+		},
+	}
+
+	mockCtrl := gomock.NewController(t)
+	ec2Mock := mock_ec2iface.NewMockEC2API(mockCtrl)
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var builder *Builder
+			if tc.expect != nil {
+				tc.expect(ec2Mock.EXPECT())
+				builder = New(tc.builder.params, WithEC2(ec2Mock))
```

Comment:
Would be good to add WithEKS() test as it is in the same file.

Reply:
Added the tests